sympy/solvers/solveset.py

```diff
@@ -2256,6 +2498,12 @@ def solvify(f, symbol, domain):
     elif isinstance(solution_set, FiniteSet):
         result = list(solution_set)
 
+    elif isinstance(solution_set, Intersection):
+        result = []
+        for argSet in solution_set.args:
+            if isinstance(argSet,FiniteSet):
+                result += list(argSet)
```

Comment:
What does this do?

For example what would `solution_set` be and what would `result` end up being?

Reply:
This converts Intersection set into a list ... (i.e. same quality as of `solve` function)